### PR TITLE
Master dev virial

### PR DESCRIPTION
**Pull request by elambros**
added virial for co2 interaction and fixed some other bugs

### CODE REVIEW

File: plugins/i-pi/src/main/driver.cpp [hunk 1]
```diff
@@ -226,6 +226,10 @@ int main(int argc, char** argv) {
             }
             // TODO check virial
             virial = std::vector<double>(9, 0.0);
+            virial = systems[0].GetVirial();
+            for (size_t i=0; i<9; i++) {
+                virial[i] /= 627.509;
```
Comment (chemphys) [inline plugins/i-pi/src/main/driver.cpp]: Virial is in energy units?

File: plugins/i-pi/src/main/driver.cpp [hunk 2]
```diff
@@ -226,6 +226,10 @@ int main(int argc, char** argv) {
             }
             // TODO check virial
             virial = std::vector<double>(9, 0.0);
```
Comment (chemphys) [inline plugins/i-pi/src/main/driver.cpp]: Maybe you can declare and initialize the virial at the begining of the main function. You can just assign it later, without need to redeclare it. I know It was done by me, though ;)

File: plugins/i-pi/test/PBC/npt/118co2/ipirun [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+#export MBX_HOME=/home/elambros/projects/mbx_virial/dev_working_master_branch/clusters_ultimate
+driver_direc=$MBX_HOME/plugins/i-pi/bin
+
+
+export OMP_NUM_THREADS=8
+
+i-pi npt_equil.xml &
+sleep 10s
+${driver_direc}/driver 118_co2.nrg mbx.json
+
```
Comment (chemphys) [inline plugins/i-pi/test/PBC/npt/118co2/ipirun]: Can you rename it to .sh? And also, delete the comment of your MBX_HOME

File: plugins/i-pi/test/PBC/npt/118co2/npt_equil.xml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+<simulation verbosity='low'>
+  <output prefix='co2_equil_test'>
+    <properties stride='1' filename='out'>  [ step, time{picosecond}, conserved{kilocal/mol}, temperature{kelvin}, potential{kelvin}, pressure_cv{megapascal}, ensemble_pressure{atmosphere} pressure_md{atmosphere}, volume ] </properties>
+    <trajectory filename='pos' stride='1' format='xyz' cell_units='angstrom'> positions{angstrom} </trajectory>
```
Comment (chemphys) [inline plugins/i-pi/test/PBC/npt/118co2/npt_equil.xml]: I would add a checkpoint directive:
```
   <output prefix=''>
      <properties filename='simulation.out' stride='10' shape='(7)'>
       [ step, time{picosecond}, conserved{kilocal/mol}, temperature{kelvin}, kinetic_cv{kelvin},
         potential{kelvin}, pressure_cv{megapascal} ]
      </properties>
      <properties filename='simulation.pot' stride='10' shape='(2)'> [ time{picosecond}, potential{kilocal/mol} ] </properties>
      <properties filename='simulation.temperature' stride='10' shape='(2)'> [ time{picosecond}, temperature{kelvin} ] </properties>
      <trajectory filename='simulation.pos' stride='10' cell_units='angstrom'>positions{angstrom}</trajectory>
      <trajectory filename='simulation.vel' format='xyz' stride='10'> velocities </trajectory>
      <checkpoint stride='10' filename='checkpoint'>39733</checkpoint>
   </output>
```

If it is not in the other tests, you can also add it!

File: plugins/i-pi/test/PBC/npt/256h2o/ipirun [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+#export MBX_HOME=/home/elambros/projects/mbx_virial/dev_working_master_branch/clusters_ultimate
+driver_direc=$MBX_HOME/plugins/i-pi/bin
+
+
+export OMP_NUM_THREADS=8
+
+i-pi npt_equil.xml &
+sleep 10s
+${driver_direc}/driver config_256.nrg mbx.json
+
```
Comment (chemphys) [inline plugins/i-pi/test/PBC/npt/256h2o/ipirun]: Same, rename to .sh and delete MBX_HOME comment

File: plugins/i-pi/test/PBC/npt/256h2o/mbx.json [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+       "max_n_eval_3b"    : 500,
+       "dipole_tolerance" : 1E-16,
+       "dipole_max_it"    : 100,
+       "dipole_method"     : "aspc",
+       "aplha_ewald_elec" : 0.25,
+       "grid_density_elec" : 2.5,
+       "spline_order_elec" : 6,
+       "aplha_ewald_disp" : 0.25, 
+       "grid_density_disp" : 2.5,
+       "spline_order_disp" : 6,
```
Comment (chemphys) [inline plugins/i-pi/test/PBC/npt/256h2o/mbx.json]: Please set ewald alpha to 0.6

File: src/tests/single_point.cpp [hunk 7]
```diff
@@ -15,7 +15,7 @@
 
 //#define NUMGRADS
 //#define PRINT_GRADS
-
+#define PRINT_VIRIAL
```
Comment (chemphys) [inline src/tests/single_point.cpp]: Comment this line

File: plugins/i-pi/src/main/driver.cpp (hunk 1 above)
Comment (elambros) [reply plugins/i-pi/src/main/driver.cpp]: yes, virial = force * distance = (kcal/(mol*A)) * A = kcal/mol

**Comment by chemphys** [comment]
Also, I just realized: is it possible that the npt examples in i-pi are missing the mbx.json?

**Pull request merged**